extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/discovery/TradfriDiscoveryParticipant.java

```diff
@@ -61,25 +61,30 @@ public ThingUID getThingUID(ServiceInfo service) {
     @Override
     public DiscoveryResult createResult(ServiceInfo service) {
 
-        if (service.getHostAddresses() != null && service.getHostAddresses().length > 0
-                && !service.getHostAddresses()[0].isEmpty()) {
-            String ip = service.getHostAddresses()[0];
+        if (service.getType().equals(getServiceType())) {
```

Comment:
Do you really need this check? I thought the framework already ensures that you are only asked for service types that you have registered for.

Reply:
I also thought so but from my experience with jmDNS this is not the case - the ```createResult``` is called for each registered participant no matter if the supported service type matches the actual one

Reply:
Indeed. But I would consider this a bug that needs to be fixed in MDNSDiscoveryService.